### PR TITLE
support providing custom path in FileExplorerView

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Preview:

https://github.com/voxel51/fiftyone/assets/25350704/eea5c9e5-1d22-4faf-a89a-84946c178a88

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/state.ts [hunk 1]
```diff
@@ -208,6 +222,20 @@ export function useFileExplorer(fsInfo, chooseMode, onChoose) {
     setCurrentDirectory(null);
   };
 
+  const handleCustomPathChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/state.ts:225]: Moving some of this logic into the path util would make it testable and could give us confidence that the various path styles supports this.

File: app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/FileExplorerView.tsx [hunk 2]
```diff
@@ -17,38 +15,6 @@ export default function FileExplorerView(props) {
...
-              reason: "Failed to get available file systems",
-              details: fsInfo.error,
-            },
-          ]}
-        />
-      </FieldWrapper>
-    );
-  }
-
-  if (fsInfo.available === false) {
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/FileExplorerView.tsx:44]: Why is this removed?

Comment (imanjra) [reply app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/FileExplorerView.tsx:44]: All of this is now moved into adornment of custom path textfield

**Review by ritch — APPROVED** [review]
